great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -295,27 +343,250 @@ def run(
 
         return expectation_suite
 
-    def self_check(self, pretty_print=True) -> dict:
-        """
-        Necessary to enable integration with `DataContext.test_yaml_config`
+    def _reconcile_variables(
+        self, variables: Optional[Dict[str, Any]] = None
+    ) -> Optional[ParameterContainer]:
+        effective_variables: Optional[ParameterContainer]
+        if variables is not None and isinstance(variables, dict):
+            variables_dict: dict = self.variables.to_dict()
+            variables_dict.update(variables)
+            effective_variables = build_parameter_container_for_variables(
+                variables_configs=variables
+            )
+        else:
+            effective_variables = self.variables
 
-        Args:
-            pretty_print: flag to turn on verbose output
+        return effective_variables
 
-        Returns:
-            Dictionary that contains RuleBasedProfiler state
+    def reconcile_rules_for_profiler(
+        self, rules: Optional[Dict[str, Dict[str, Any]]] = None
+    ) -> List[Rule]:
+        if rules is None:
+            rules = {}
```

Comment:
Could we please get docstrings for this function and anything else public? Just trying to enforce the Google style docstring noted in our style guide on Confluence.

Reply:
@cdkini Will add docstrings in the next PR when adding tests (there is a `TODO` already for that).  For now, we need to get the commented tests to work and get this refactoring PR merged ASAP.

Reply:
Okay but please try to ensure this doesn't slip through the cracks 😄 at a minimum, I'd say to docstring the more complex functions (like this reconciliation). The more obvious ones don't necessarily need it but I'd love more documentation if at all possible.